### PR TITLE
Jawi keyboard

### PR DESCRIPTION
**Pull request by bennylin**
For https://meta.wikimedia.org/wiki/WILMA_Proofread-a-thon_2023

### CODE REVIEW

**Comment by bennylin** [comment]
Hi, I don't know what the error was. Can anyone help me decipher the error?

**Comment by bennylin** [comment]
Since Amir is preoccupied, could @santhoshtr probably help with this time-sensitive PR?

**Comment by Abijeet** [comment]
Closed and re-opened to re-run CI

File: test/jquery.ime.test.fixtures.js [hunk 1]
```diff
@@ -2211,6 +2211,37 @@ var palochkaVariants = {
...
+			{ input: 'q', output: 'ق', description: 'Jawi qaf' },
+			{ input: 'r', output: 'ر', description: 'Jawi ra' },
+			{ input: 's', output: 'س', description: 'Jawi sin' },
+			{ input: 't', output: 'ت', description: 'Jawi ta' },
+			{ input: 'u', output: 'ُ', description: 'Jawi dhammah -u' },
+			{ input: 'v', output: 'ؤ', description: 'Jawi va' },
+			{ input: 'w', output: 'و', description: 'Jawi wau' },
+			{ input: 'x', output: 'ء', description: 'Jawi hamza' },
+			{ input: 'y', output: 'ي', description: 'Jawi ya' },
+			{ input: 'z', output: 'ز', description: 'Jawi zai' }
```
Comment (Abijeet) [inline test/jquery.ime.test.fixtures.js:2243]: Closing `]` (started on line 2217) missing which is causing the test failures.

**Review by Abijeet — APPROVED** [review]

File: rules/id/id-jawi.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		description: 'Jawi keyboard for Indonesian Wikisource',
+		date: '2023-09-21',
+		URL: 'https://www.mediawiki.org/wiki/Help:Extension:UniversalLanguageSelector/Input_methods/jawi-keyboard',
+		author: 'Benny Lin, Pungkas, Michael',
+		license: 'GPLv3',
+		version: '1.0',
+		contextLength: 3,
+		maxKeyLength: 3,
+		patterns: [
+			[ '\\\\([A-Za-z\\>_~\\.0-9])', '$1' ], //
```
Comment (Nikerabbit) [inline rules/id/id-jawi.js:15]: empty comment? I'm not sure what this line does. Identity transform for `\` followed by a letter? Shouldn't `$1` be a third parameter.

File: test/jquery.ime.test.fixtures.js [hunk 3]
```diff
@@ -2211,6 +2211,38 @@ var palochkaVariants = {
 			{ input: 'a~/e~\\i~^o~-u~v', output: 'áèîōǔ', description: 'Ibibio tilde áèîōǔ' }
 		]
 	},
+	{
+		description: 'Jawi test',
+		inputmethod: 'id-jawi',
+		tests: [
```
Comment (Nikerabbit) [inline test/jquery.ime.test.fixtures.js:2217]: Would be nice to have test cases for the more complex rules.

Comment (bennylin) [reply test/jquery.ime.test.fixtures.js:2217]: On it

File: rules/id/id-jawi.js (hunk 2 above)
Comment (bennylin) [reply rules/id/id-jawi.js]: changed. thanks.

**Comment by bennylin** [comment]
@Nikerabbit and @Abijeet could you guys take a look. Thanks.

File: rules/id/id-jawi.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+﻿( function ( $ ) {
+	'use strict';
+	var jawiKeyboard = {
+		id: 'id-jawi',
+		name: 'Jawi keyboard',
+		description: 'Jawi keyboard for Indonesian Wikisource',
+		date: '2023-09-21',
+		URL: 'https://www.mediawiki.org/wiki/Help:Extension:UniversalLanguageSelector/Input_methods/jawi-keyboard',
```
Comment (santhoshtr) [inline rules/id/id-jawi.js:8]: Need to create this page. Not a blcoker for the PR though

File: rules/id/id-jawi.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+		author: 'Benny Lin, Pungkas, Michael',
+		license: 'GPLv3',
+		version: '1.0',
+		contextLength: 3,
```
Comment (santhoshtr) [inline rules/id/id-jawi.js:12]: As far as I understand this input method is a 1:1 input method. The output of key strokes does not depend on previous input(context). hence concxtLength can be 0. Please validate

File: rules/id/id-jawi.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+		maxKeyLength: 3,
```
Comment (santhoshtr) [inline rules/id/id-jawi.js:13]: Is it 3? 
Please check with the definition of field at https://github.com/wikimedia/jquery.ime/tree/master/rules#input-method-definition

File: rules/id/id-jawi.js (hunk 4 above)
Comment (bennylin) [reply rules/id/id-jawi.js:8]: done
https://www.mediawiki.org/wiki/Help:Extension:UniversalLanguageSelector/Input_methods/jawi-keyboard

File: rules/id/id-jawi.js (hunk 5 above)
Comment (bennylin) [reply rules/id/id-jawi.js:12]: it does depend on the previous input, as in 'gh', 'th', 'kh' characters

File: rules/id/id-jawi.js (hunk 4 above)
Comment (srish) [reply rules/id/id-jawi.js:8]: Perhaps the on-wiki documentation need to be updated as well to reflect the changes in the PR.

Comment (bennylin) [reply rules/id/id-jawi.js:8]: done

**Pull request merged**